### DISCUSSION TITLE
Under which circumstances would psycopg2 send an ETB (23 , 0x17) message?

### DISCUSSION THREAD

**Discussion by cpaulik** [psycopg/psycopg2#1678] [Q&A] [answered]
We are seeing errors in our pgbouncer installation like this

```
unknown pkt from client: 23/0x17
```

As far as I can tell this comes when Byte1 in the message is 23 instead of a recognized type [here](https://github.com/pgbouncer/pgbouncer/blob/e913a15592c5d03008fc60b15be692e310e37d21/src/client.c#L1104) according to the spec [here](https://www.postgresql.org/docs/current/protocol-message-formats.html)

So I was trying to find where this would happen in psycopg2 but couldn't find it.

**Comment by dvarrazzo** [comment, marked as answer]
In psycopg2, nothing: psycopg doesn't send messages on wire, it only uses the libpq.

I think that the libpq is not to blame either, because the byte1 it sends I think is always an ascii readable char, and 0x17 is not.

So I think either it is some library used on connection such as gssapi or openssl, or, well, no idea.

**Reply by cpaulik** [reply]
Thanks for the pointers. I'll dig further down the stack then.
